crates/conversions/src/felt252.rs

```diff
@@ -69,6 +83,47 @@ impl FromShortString<Felt252> for Felt252 {
     }
 }
 
+pub trait TryInferFormat: Sized {
+    /// Parses value from `hex string`, `dec string` and `quotted cairo shortstring`
+    fn infer_format_and_parse(value: &str) -> Result<Vec<Self>, ParseFeltError>;
+}
+
+impl<T> TryInferFormat for T
+where
+    T: From<Felt252>,
+{
+    fn infer_format_and_parse(mut value: &str) -> Result<Vec<Self>, ParseFeltError> {
+        let mut result = Err(ParseFeltError);
+
+        let is_short_string = value.starts_with('\'') && value.ends_with('\'');
```

Comment:
is this assumption safe? shouldn't we rather check the actual length/size instead?

Reply:
ok I now see that this is assumed in snforge_std docs, but still should be only depend on that? seems rather risky to me

Reply:
`Felt252::from_short_string` checks it already so no need to duplicate logic here.
```rs
fn from_short_string(short_string: &str) -> Result<Felt252, ParseFeltError> {
    if short_string.len() <= 31 && short_string.is_ascii() {
        Ok(Felt252::from_bytes_be(short_string.as_bytes()))
    } else {
        Err(ParseFeltError)
    }
}
```
This only decides if apply `Felt252::from_short_string` or not

Reply:
I see your point, but it just seems counter-intuitive to me that `is_short_string` does not check if something is actually a short string, but `from_short_string` that is being called later in the code does.